account_asset_management/tests/test_account_asset_management.py

```diff
@@ -186,7 +186,24 @@ def test_01_nonprorata_basic(self):
         self.assertEqual(ict0.value_depreciated, 500)
         self.assertEqual(ict0.value_residual, 1000)
         vehicle0.validate()
-        vehicle0.depreciation_line_ids[1].create_move()
+        created_move_ids = vehicle0.depreciation_line_ids[1].create_move()
+        for move_id in created_move_ids:
+            move = self.env["account.move"].browse(move_id)
+            move_lines_w_analytic_data = move.line_ids.filtered(
+                lambda line: line.analytic_account_id ==
+                self.env.ref("analytic.analytic_administratif") and
+                line.analytic_tag_ids ==
+                self.env.ref("analytic.tag_contract")
+            )
+            self.assertTrue(move_lines_w_analytic_data)
+            self.assertEquals(
+                move_lines_w_analytic_data,
+                move.line_ids.filtered(
+                    lambda line: line.account_id == self.env.ref("account.a_expense")
+                    # Filter for expense lines
+                    # (associated with the profile's depreciation account)
+                )
+            )
```

Comment:
Cleaner code IMO is:

```suggestion
            expense_line = move.line_ids.filtered(
                lambda line: line.account_id == self.env.ref("account.a_expense")
            )
            self.assertEqual(
                expense_line.analytic_account_id,
                self.env.ref("analytic.analytic_administratif"),
            )
            self.assertEqual(
                expense_line.analytic_tag_ids,
                self.env.ref("analytic.tag_contract")
            )
```

Reply:
Thanks :+1: 
Much better, indeed :smile: